### DISCUSSION TITLE
What is the returnUri for Oauth2 authentication requests

### DISCUSSION THREAD

**Discussion by kale1d0code** [grafana/grafana-infinity-datasource#858] [Q&A] [answered]
I am trying to setup an Oauth2 connection to the xero API.
the xero API requires a return Uri. I am unsure what this should be set to and I am unable to find any information online so to what the return Uri would be for the grafana-infinity-datasource.

Does anyone have this information?

**Comment by kale1d0code** [comment, marked as answer]
Found the answer here
there is no request/redirect uri

https://github.com/grafana/grafana-infinity-datasource/discussions/310

**Reply by Parmenion49** [reply]
Hi
May I ask what you did to get around this issue? I am only creating a few test dashboards atm and is currently using postman to do the authorisation and then using the token provided in Infinity as an API Key value pair. 
Regards
P

**Reply by kale1d0code** [reply]
Oh I never did.
You can't use Xero's tenanted API with infinity.
the tenanted api uses an authentication workflow which requires human interaction.
